File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        currMin = prices[0]
        maxProfit = 0

        for i in range(1, len(prices)):
            if prices[i] < currMin:
                currMin = prices[i]
            else:
                maxProfit = max(maxProfit, prices[i] - currMin)

        return maxProfit
            

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        if prices.count == 1 {
            return 0
        }

        var currMin = prices[0]
        var maxProfit = 0

        for i in 1..<prices.count {
            if prices[i] < currMin {
                currMin = prices[i]
            } else if prices[i] - currMin > maxProfit {
                maxProfit = prices[i] - currMin
            }
        }

        return maxProfit
    }
}